vilebot/src/main/java/com/oldterns/vilebot/handlers/user/KarmaTransfer.java

```diff
@@ -0,0 +1,289 @@
+/**
+ * Copyright (C) 2020 Oldterns
+ *
+ * This file may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.oldterns.vilebot.handlers.user;
+
+import com.oldterns.vilebot.Vilebot;
+import com.oldterns.vilebot.db.KarmaDB;
+import com.oldterns.vilebot.util.BaseNick;
+import org.pircbotx.hooks.ListenerAdapter;
+import org.pircbotx.hooks.events.MessageEvent;
+import org.pircbotx.hooks.types.GenericMessageEvent;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KarmaTransfer
+    extends ListenerAdapter
+{
+    private static final Pattern transferPattern = Pattern.compile( "^!transfer\\s+(\\S+)\\s+([0-9]+)" );
+
+    private static final Pattern cancelTransferPattern = Pattern.compile( "^!transfercancel" );
+
+    private static final Pattern acceptTransferPattern = Pattern.compile( "^!accept" );
+
+    private static final Pattern rejectTransferPattern = Pattern.compile( "^!reject" );
+
+    private KarmaTransaction currentTransaction;
+
+    private final Object currentTransferMutex = new Object();
+
+    private static ExecutorService timer = Executors.newScheduledThreadPool( 1 );
+
+    private static final long TIMEOUT = 30000L;
+
+    @Override
+    public void onGenericMessage( final GenericMessageEvent event )
+    {
+        String text = event.getMessage();
+
+        Matcher transferMatcher = transferPattern.matcher( text );
+        Matcher cancelTransferMatcher = cancelTransferPattern.matcher( text );
+        Matcher acceptTransferMatcher = acceptTransferPattern.matcher( text );
+        Matcher rejectTransferMatcher = rejectTransferPattern.matcher( text );
+
+        if ( transferMatcher.matches() )
+            transferKarma( event, transferMatcher );
+        if ( cancelTransferMatcher.matches() )
+            cancelTransfer( event );
+        if ( acceptTransferMatcher.matches() )
+            acceptTransfer( event );
+        if ( rejectTransferMatcher.matches() )
+            rejectTransfer( event );
+    }
+
+    private void transferKarma( GenericMessageEvent event, Matcher transferMatcher )
+    {
+        // Prevent users from transferring karma outside of #thefoobar
+        if ( !( event instanceof MessageEvent )
+            || !( (MessageEvent) event ).getChannel().getName().equals( Vilebot.getConfig().get( "ircChannel1" ) ) )
+        {
+            event.respondWith( "You must be in " + Vilebot.getConfig().get( "ircChannel1" )
+                + " to make or accept wagers." );
```

Comment:
I am 99% sure this error message doesn't make sense since it talking about !roll in `KarmaTransfer`

Reply:
Haha oops, forgot to change it. Nice catch!